crates/viewer/re_viewer_context/src/cache/video_stream_cache.rs

```diff
@@ -0,0 +1,587 @@
+                re_log::warn_once!(
+                    "Out of order logging on video streams is not supported. Ignoring any out of order samples."
+                );
+                return Ok(());
+            }
+        }
+        None => {
+            // This chunk doesn't have any data on this timeline.
+            return Ok(());
+        }
+    }
+
+    // The underlying data within a chunk is logically a Vec<Vec<Blob>>,
+    // where the inner Vec always has a len=1, because we're dealing with a "mono-component"
+    // (each VideoStream has exactly one VideoSample instance per time)`.
+    //
+    // Because of how arrow works, the bytes of all the blobs are actually sequential in memory (yay!) in a single buffer,
+    // what you call values below (could use a better name btw).
+    //
+    // We want to figure out the byte offsets of each blob within the arrow buffer that holds all the blobs,
+    // i.e. get out a Vec<ByteRange>.
+    let inner_list_array = raw_array
+        .downcast_array_ref::<arrow::array::ListArray>()
+        .ok_or(VideoStreamProcessingError::InvalidVideoSampleType(
+            raw_array.data_type().clone(),
+        ))?;
+    let values = inner_list_array
+        .values()
+        .downcast_array_ref::<arrow::array::PrimitiveArray<arrow::array::types::UInt8Type>>()
+        .ok_or(VideoStreamProcessingError::InvalidVideoSampleType(
+            raw_array.data_type().clone(),
+        ))?;
+    let values = values.values().inner();
+
+    let offsets = inner_list_array.offsets();
+    let lengths = offsets.lengths().collect::<Vec<_>>();
+
+    let buffer_index = chunk_buffers.next_index();
+    let sample_base_idx = samples.next_index();
+
+    // Extract sample metadata.
+    samples.extend(
+        chunk
+            .iter_component_offsets(&sample_descr)
+            .zip(chunk.iter_component_indices(&timeline, &sample_descr))
+            .filter_map(move |((idx, len), (time, _row_id))| {
+                if len == 0 {
+                    // Ignore empty samples.
+                    return None;
+                }
+                if len != 1 {
+                    re_log::warn_once!(
+                        "Expected only a single VideoSample per row (it is a mono-component)"
+                    );
+                    return None;
+                }
+
+                let sample_idx = sample_base_idx + idx;
+                let byte_offset = offsets[idx] as usize;
+                let byte_length = lengths[idx];
+
+                // Note that the conversion of this time value is already handled by `VideoDataDescription::timescale`:
+                // For sequence time we use a scale of 1, for nanoseconds time we use a scale of 1_000_000_000.
+                let decode_timestamp = re_video::Time(time.as_i64());
+
+                // Samples within a chunk are expected to be always in order.
+                debug_assert!(decode_timestamp > previous_max_presentation_timestamp);
```

Comment:
I think this is a bad assumption. As noted above, chunks are sorted by `RowId`, but cannot be sorted on all timelines. We should have a test that actively log a video backwards and in a random order so make sure we don't crash in those cases

Reply:
but we are iterating on them in order with those iterators above? 🤔
sounds like we should huddle about this, but going to be hard to squeeze that in. I can at the _very least_ adjust those comments to not lie about chunks!